internal/obsreportconfig/obsreportconfig.go

```diff
@@ -103,6 +116,26 @@ func allViews() []*view.View {
 	return views
 }
 
+func receiverViews() []*view.View {
+	if featuregate.GetRegistry().IsEnabled(UseOtelForInternalMetricsfeatureGateID) {
```

Comment:
kinda misc, but I've added featuregate check on the OC receiver views creation.

Reply:
why? I don't think we should code this way, the contract is to register these views only if `UseOtelForInternalMetricsfeatureGateID`, then we protect the high level function we don't have to protect any sub-function.

Reply:
Because if we want to use `prometheus.Registry` as the bridge, we don't want OC to emit metrics that have being migrated to otel-go if the featuregate is enabled, but we would want all the others non-migrated oc metrics to be exported as normal, hence I just removed the `receiver` metrics for now.

Once all components of the core gets instrumented with otel-go, we can move this to be broader and not register any view at all when the featuregate is enabled, but for now I would suggest to just remove it component per component.